src/net/wooga/jenkins/pipeline/BuildUtils.groovy

```diff
@@ -0,0 +1,56 @@
+package net.wooga.jenkins.pipeline
+
+import org.jenkinsci.plugins.pipeline.modeldefinition.Utils
+
+class BuildVersion {
+    final String version
+    final Boolean optional
+
+    BuildVersion(version, optional) {
+        this.version = version
+        this.optional = optional
+    }
+    
+    @Override
+    String toString() {
+        return "version : ${version}, optional : ${optional}"
+    }
+}
+
+// For each entry in a versions map, if they are not in the standard format
+// we process them into the newer format that supports optional arguments per version
+// (such as if it's optional, etc)
+// Example: '3001.x' -> [ version : '3001.x', optional : false ]
+def parseVersions(versions) {
+
+    final versionKey = "version"
+    final optionalKey = "optional"
+
+    versions = versions.collect { $it -> 
```

Comment:
You don't have to name it `$it` 
`it` is the default binding from groovy when the closure only gets one item assigned as a shortform.
Here you can simple name the parameter how ever you like.

Reply:
Oh, that's good to know!